src/nvim/ex_cmds.c

```diff
@@ -321,12 +321,10 @@ static int sort_compare(const void *s1, const void *s2)
     /* We need to copy one line into "sortbuf1", because there is no
      * guarantee that the first pointer becomes invalid when obtaining the
      * second one. */
-    STRNCPY(sortbuf1, ml_get(l1.lnum) + l1.start_col_nr,
+    xstrlcpy(sortbuf1, (char *)ml_get(l1.lnum) + l1.start_col_nr,
```

Comment:
`xstrlcpy()` uses an extra call to `strlen()`.  Is it necessary to use `STRNCPY()` or `xstrlcpy()` here?

We are copying part of line `l1.lnum` and know that `sortbuf1` has always enough space for the whole line + `NUL`. So instead of using `STRNCPY()` and `xstrlcpy()`, why not use `memcpy()` and avoid the overhead?
```c
    memcpy(sortbuf1, (char *)ml_get(l1.lnum) + l1.start_col_nr, l1.end_col_nr - l1.start_col_nr);
    sortbuf1[l1.end_col_nr - l1.start_col_nr] = 0;
    memcpy(sortbuf2, (char *)ml_get(l2.lnum) + l2.start_col_nr, l2.end_col_nr - l2.start_col_nr);
    sortbuf2[l2.end_col_nr - l2.start_col_nr] = 0;
```